app/controllers/lti_nrps_controller.rb

```diff
@@ -84,11 +84,72 @@ def send_nrps_request
     lcd.last_synced = DateTime.current
     lcd.save
 
-    render json: members.as_json
+    # Update the roster with the retrieved set of members
+    @cuds = parse_members_data(lcd, members.as_json)
+    @sorted_cuds = @cuds.sort_by { |cud| cud[:color] || "z" }.reverse
+    @course = lcd.course
   end
 
 private
 
+  def parse_members_data(lcd, members_data)
+    cuds = CourseUserDatum.where(course_id: lcd.course_id, instructor: false,
+                                 course_assistant: false).to_set
+    email_to_cud = {}
+    cuds.each do |cud|
+      email_to_cud[cud.user.email] = cud
+    end
+
+    cud_view = []
+    members_data.each do |user_data|
+      next unless user_data["roles"].include? "http://purl.imsglobal.org/vocab/lis/v2/membership#Learner"
+
+      next if user_data.key?("status") && user_data["status"] != "Active"
+
+      cud_data = {}
+      # Normalize email
+      user_data["email"].downcase!
+
+      user = User.find_by(email: user_data["email"])
+      cud_data[:color] = if user.nil? || lcd.course.course_user_data.find_by(user_id: user.id).nil?
+                           "green"
+                         else
+                           "black"
+                         end
+      cud_data[:email] = user_data["email"]
```

Comment:
btw if this user exists, we should make sure to populate the other fields (e.g. school, major, year, grade_policy, etc.), because right now these fields are blank when a sync occurs, which then overwrites the previous data, when we should preserve the fields we don't touch.

Reply:
The logic from `courses_controller.rb#write_cuds` can probably be reused here

Reply:
I'll make sure to preserve the other fields. The logic from `courses_controller.rb#wrtie_cuds` I think is already being used when the updateRoster confirm button is pressed.